# DKT Graph and Runner
Test the remodeled DKT graph and runner.

In [1]:
from Hyperparameter import *
from DataGenerator import *
from DKTGraph import *
from DKTRunner import *

hps = Hyperparameter(dataset_file='Assistments/assistment_dirty_for_dkt.csv',
                     split_file='Assistments/dirty_split.csv',
                     batch_size=100, learning_rate=0.001, dropout_keep=0.5, lstm_cell='standard',
                     num_epochs=200, padding_scheme='end_0', pad_to=0)
#hps = Hyperparameter(dataset_file='synthetic/set_2.csv',
#                     split_file='split_50%train.csv', learning_rate=0.01,
#                     lstm_cell='standard', clipping_norm=2.0)
data_generator = DataGenerator(hps)

total: 4108 student sequences
3284 sequences, 421834 records for train
824 sequences, 103591 records for test
all batch generated


# Oscillation comes from bad hyperparameters?
Dirty Assistment: 0.855 (high learning rate, dropout => Quick Convergence, but saw oscillation; low learning rate, no dropout => stable); same with the original version (0.850); discard 1-trial students 0.83

Corrected Assistment: 0.806 (lower than the original version 0.825)

124 skill Corrected Assistment: 0.800, discard length=1, drop to 0.78

peephole: doesn't seem to boost performance, need small learning rate

Synthetic dataset: Set9 0.81~0.82 with skill; 0.80 no skill; Set2 0.826 with skill ; 0.825 no skill

In [2]:
dkt_graph = DKTGraph(hps, data_generator)
dkt_runner = DKTRunner()
dkt_runner.train(hps, dkt_graph, data_generator)

epoch 0: loss = 0.67779864145
Train AUC = 0.596914124629
Test AUC = 0.612984773692
Test accuracy = 0.653493064069    
epoch 1: loss = 0.644609133403
Train AUC = 0.629014231788
Test AUC = 0.648620844439
Test accuracy = 0.68990549372    
epoch 2: loss = 0.633431653182
Train AUC = 0.633354598796
Test AUC = 0.660889358992
Test accuracy = 0.695591315848    
epoch 3: loss = 0.626582647815
Train AUC = 0.649454919922
Test AUC = 0.677505276332
Test accuracy = 0.698323213407    
epoch 4: loss = 0.623809736786
Train AUC = 0.654045391735
Test AUC = 0.682973670239
Test accuracy = 0.700620710293    
epoch 5: loss = 0.604551015478
Train AUC = 0.707506302791
Test AUC = 0.744220920587
Test accuracy = 0.743539496674    
epoch 6: loss = 0.590117535808
Train AUC = 0.744354320476
Test AUC = 0.768082243705
Test accuracy = 0.739485090404    
epoch 7: loss = 0.58037411715
Train AUC = 0.775805391683
Test AUC = 0.784535591825
Test accuracy = 0.74712088888    
epoch 8: loss = 0.575735895923
Train AUC = 0.7849303

In [3]:
for batch in data_generator.train_batches:
    print batch.maxlen

33
8
17
21
62
690
249
126
30
10
15
24
973
195
5
13
88
8214
12
2
73
27
7
105
39
19
45
374
155
6
52
10
3


# Assitments Dirty Data
## my random split
epoch 0: loss = 0.67779864145
Train AUC = 0.596914124629
Test AUC = 0.612984773692
Test accuracy = 0.653493064069    
epoch 1: loss = 0.644609133403
Train AUC = 0.629014231788
Test AUC = 0.648620844439
Test accuracy = 0.68990549372    
epoch 2: loss = 0.633431653182
Train AUC = 0.633354598796
Test AUC = 0.660889358992
Test accuracy = 0.695591315848    
epoch 3: loss = 0.626582647815
Train AUC = 0.649454919922
Test AUC = 0.677505276332
Test accuracy = 0.698323213407    
epoch 4: loss = 0.623809736786
Train AUC = 0.654045391735
Test AUC = 0.682973670239
Test accuracy = 0.700620710293    
epoch 5: loss = 0.604551015478
Train AUC = 0.707506302791
Test AUC = 0.744220920587
Test accuracy = 0.743539496674    
epoch 6: loss = 0.590117535808
Train AUC = 0.744354320476
Test AUC = 0.768082243705
Test accuracy = 0.739485090404    
epoch 7: loss = 0.58037411715
Train AUC = 0.775805391683
Test AUC = 0.784535591825
Test accuracy = 0.74712088888    
epoch 8: loss = 0.575735895923
Train AUC = 0.784930337755
Test AUC = 0.794325068671
Test accuracy = 0.752150283326    
epoch 9: loss = 0.572805867051
Train AUC = 0.789928031271
Test AUC = 0.803814854876
Test accuracy = 0.763087526909    
epoch 10: loss = 0.569766428434
Train AUC = 0.793774515654
Test AUC = 0.804275074383
Test accuracy = 0.764342462183    
epoch 11: loss = 0.566212671273
Train AUC = 0.798359937137
Test AUC = 0.809562750756
Test accuracy = 0.767836974255    
epoch 12: loss = 0.562176922957
Train AUC = 0.801614494619
Test AUC = 0.812353000278
Test accuracy = 0.770202044579    
epoch 13: loss = 0.560241701025
Train AUC = 0.800115578089
Test AUC = 0.815981526986
Test accuracy = 0.773098049058    
epoch 14: loss = 0.555999006286
Train AUC = 0.807720571083
Test AUC = 0.816522897931
Test accuracy = 0.773841356875    
epoch 15: loss = 0.552708186887
Train AUC = 0.808515701859
Test AUC = 0.819343811716
Test accuracy = 0.775270052418    
epoch 16: loss = 0.551223460472
Train AUC = 0.811152864777
Test AUC = 0.819839024064
Test accuracy = 0.775501732776    
epoch 17: loss = 0.548161975362
Train AUC = 0.813416585019
Test AUC = 0.822306398296
Test accuracy = 0.778523230783    
epoch 18: loss = 0.547732898683
Train AUC = 0.81377630618
Test AUC = 0.822425015326
Test accuracy = 0.777635122742    
epoch 19: loss = 0.544108122587
Train AUC = 0.815361271646
Test AUC = 0.824600277723
Test accuracy = 0.778909364713    
epoch 20: loss = 0.542440954483
Train AUC = 0.816118748477
Test AUC = 0.824545026934
Test accuracy = 0.779932619629    
epoch 21: loss = 0.541299350334
Train AUC = 0.816397257934
Test AUC = 0.825720452655
Test accuracy = 0.780367020301    
epoch 22: loss = 0.539711837516
Train AUC = 0.818028959173
Test AUC = 0.826157277913
Test accuracy = 0.78076280758    
epoch 23: loss = 0.537845064293
Train AUC = 0.819227191082
Test AUC = 0.826181298102
Test accuracy = 0.7809269145    
epoch 24: loss = 0.536010638331
Train AUC = 0.820340960921
Test AUC = 0.826293865108
Test accuracy = 0.781062061376    
epoch 25: loss = 0.534727065852
Train AUC = 0.820076692595
Test AUC = 0.828772623054
Test accuracy = 0.781892249327    
epoch 26: loss = 0.534994831591
Train AUC = 0.821322922494
Test AUC = 0.828519342178
Test accuracy = 0.782268729909    
epoch 27: loss = 0.532984797702
Train AUC = 0.822502781881
Test AUC = 0.829687346568
Test accuracy = 0.783108571208    
epoch 28: loss = 0.530812988679
Train AUC = 0.822408679016
Test AUC = 0.830184376694
Test accuracy = 0.783967719203    
epoch 29: loss = 0.529665942445
Train AUC = 0.824029123899
Test AUC = 0.829723361895
Test accuracy = 0.783600891969    
epoch 30: loss = 0.529019484014
Train AUC = 0.824190849186
Test AUC = 0.830242571339
Test accuracy = 0.783890492417    
epoch 31: loss = 0.52733034618
Train AUC = 0.824741599736
Test AUC = 0.82975095056
Test accuracy = 0.783649158711    
epoch 32: loss = 0.526706569123
Train AUC = 0.824214327276
Test AUC = 0.831207909152
Test accuracy = 0.784479346661    
epoch 33: loss = 0.526082691821
Train AUC = 0.825992348463
Test AUC = 0.830434060722
Test accuracy = 0.783571931925    
epoch 34: loss = 0.524207171166
Train AUC = 0.826493061945
Test AUC = 0.832226459159
Test accuracy = 0.784595186841    
epoch 35: loss = 0.522729491646
Train AUC = 0.827329184414
Test AUC = 0.831562974187
Test accuracy = 0.78465310693    
epoch 36: loss = 0.523256026434
Train AUC = 0.824220529928
Test AUC = 0.833244958141
Test accuracy = 0.785232307826    
epoch 37: loss = 0.521452861302
Train AUC = 0.828340778622
Test AUC = 0.83273898604
Test accuracy = 0.785213001129    
epoch 38: loss = 0.521153579156
Train AUC = 0.827703497344
Test AUC = 0.833234707102
Test accuracy = 0.785106814299    
epoch 39: loss = 0.520578429554
Train AUC = 0.8268792909
Test AUC = 0.824194724192
Test accuracy = 0.78137096852    
epoch 40: loss = 0.523454483711
Train AUC = 0.824908375584
Test AUC = 0.833278211061
Test accuracy = 0.785483294881    
epoch 41: loss = 0.519070279418
Train AUC = 0.828161077571
Test AUC = 0.833871327881
Test accuracy = 0.786033535732    
epoch 42: loss = 0.518120108229
Train AUC = 0.828024550945
Test AUC = 0.833759422885
Test accuracy = 0.785724628587    
epoch 43: loss = 0.517292971864
Train AUC = 0.829412591145
Test AUC = 0.835117848425
Test accuracy = 0.785705321891    
epoch 44: loss = 0.516404393947
Train AUC = 0.83010244099
Test AUC = 0.834261539712
Test accuracy = 0.785628095105    
epoch 45: loss = 0.514732974045
Train AUC = 0.830701789077
Test AUC = 0.834687631391
Test accuracy = 0.785830815418    
epoch 46: loss = 0.513341170369
Train AUC = 0.830238219348
Test AUC = 0.835153196024
Test accuracy = 0.786438976359    
epoch 47: loss = 0.513528908744
Train AUC = 0.830971559767
Test AUC = 0.835814706623
Test accuracy = 0.786583776583    
epoch 48: loss = 0.511812070102
Train AUC = 0.83146853005
Test AUC = 0.833496719473
Test accuracy = 0.785714975239    
epoch 49: loss = 0.511632288044
Train AUC = 0.830950936991
Test AUC = 0.836271630595
Test accuracy = 0.786014229035    
epoch 50: loss = 0.512567372033
Train AUC = 0.830733716444
Test AUC = 0.835638887964
Test accuracy = 0.786236256045    
epoch 51: loss = 0.509379055464
Train AUC = 0.831623404183
Test AUC = 0.83548410165
Test accuracy = 0.786410016314    
epoch 52: loss = 0.508714316469
Train AUC = 0.832197965869
Test AUC = 0.835361046514
Test accuracy = 0.786506549797    
epoch 53: loss = 0.507191027656
Train AUC = 0.832767688194
Test AUC = 0.836577554248
Test accuracy = 0.78698921721    
epoch 54: loss = 0.507300728198
Train AUC = 0.832275136947
Test AUC = 0.836384527066
Test accuracy = 0.786844416986    
epoch 55: loss = 0.506636763161
Train AUC = 0.833057447415
Test AUC = 0.835265544632
Test accuracy = 0.785975615642    
epoch 56: loss = 0.505392417763
Train AUC = 0.831964204253
Test AUC = 0.836083779628
Test accuracy = 0.786854070334    
epoch 57: loss = 0.505059754307
Train AUC = 0.833566647357
Test AUC = 0.836052605991
Test accuracy = 0.787037483951    
epoch 58: loss = 0.504318855929
Train AUC = 0.832796076023
Test AUC = 0.836444374537
Test accuracy = 0.787056790648    
epoch 59: loss = 0.502443795854
Train AUC = 0.834161825971
Test AUC = 0.83651578387
Test accuracy = 0.786940950469    
epoch 60: loss = 0.502759935278
Train AUC = 0.834183280087
Test AUC = 0.835343332796
Test accuracy = 0.786429323011    
epoch 61: loss = 0.50252949469
Train AUC = 0.834061539289
Test AUC = 0.837375987885
Test accuracy = 0.787664951588    
epoch 62: loss = 0.500048518181
Train AUC = 0.835102427097
Test AUC = 0.836983224785
Test accuracy = 0.787076097344    
epoch 63: loss = 0.499588912184
Train AUC = 0.835040185412
Test AUC = 0.836057792104
Test accuracy = 0.786139722563    
epoch 64: loss = 0.499035630262
Train AUC = 0.833837402235
Test AUC = 0.836732234332
Test accuracy = 0.786834763638    
epoch 65: loss = 0.498291202567
Train AUC = 0.835619777875
Test AUC = 0.837486565561
Test accuracy = 0.788369646012    
epoch 66: loss = 0.49684795376
Train AUC = 0.836179126357
Test AUC = 0.836973678424
Test accuracy = 0.786940950469    
epoch 67: loss = 0.494637033253
Train AUC = 0.83601256371
Test AUC = 0.837066735267
Test accuracy = 0.787346391096    
epoch 68: loss = 0.49537618955
Train AUC = 0.836022227827
Test AUC = 0.836513261414
Test accuracy = 0.787510498016    
epoch 69: loss = 0.494019135381
Train AUC = 0.836346330082
Test AUC = 0.837545987893
Test accuracy = 0.787780791768    
epoch 70: loss = 0.492904647733
Train AUC = 0.836950497041
Test AUC = 0.83377941559
Test accuracy = 0.786390709618    
epoch 71: loss = 0.493146585696
Train AUC = 0.836943725485
Test AUC = 0.837370073782
Test accuracy = 0.787858018554    
epoch 72: loss = 0.491470484119
Train AUC = 0.837514528766
Test AUC = 0.837198701807
Test accuracy = 0.787385004489    
epoch 73: loss = 0.491799851259
Train AUC = 0.836556927499
Test AUC = 0.83799264873
Test accuracy = 0.78793524534    
epoch 74: loss = 0.491468269717
Train AUC = 0.835809170662
Test AUC = 0.836196939144
Test accuracy = 0.787066443996    
epoch 75: loss = 0.490404073939
Train AUC = 0.835901346192
Test AUC = 0.836517105911
Test accuracy = 0.787848365205    
epoch 76: loss = 0.488835791747
Train AUC = 0.838066471756
Test AUC = 0.83715951147
Test accuracy = 0.787452577927    
epoch 77: loss = 0.48690568588
Train AUC = 0.838852735534
Test AUC = 0.837586303867
Test accuracy = 0.788022125474    
epoch 78: loss = 0.485299515905
Train AUC = 0.839169512203
Test AUC = 0.837325763974
Test accuracy = 0.787616684847    
epoch 79: loss = 0.485694315397
Train AUC = 0.839236099019
Test AUC = 0.836304580882
Test accuracy = 0.78715332413    
epoch 80: loss = 0.486256547046
Train AUC = 0.83875164177
Test AUC = 0.837110249783
Test accuracy = 0.787085750693    
epoch 81: loss = 0.485533249198
Train AUC = 0.837791746233
Test AUC = 0.836579999727
Test accuracy = 0.786728576807    
epoch 82: loss = 0.483942742601
Train AUC = 0.838076130591
Test AUC = 0.835507476984
Test accuracy = 0.787220897568    
epoch 83: loss = 0.482989385273
Train AUC = 0.838404217761
Test AUC = 0.837187217801
Test accuracy = 0.787819405161    
epoch 84: loss = 0.482066120162
Train AUC = 0.838886285128
Test AUC = 0.837596599334
Test accuracy = 0.788427566101    
epoch 85: loss = 0.479941490925
Train AUC = 0.840566414349
Test AUC = 0.836261468851
Test accuracy = 0.787201590872    
epoch 86: loss = 0.479862084894
Train AUC = 0.8390081369
Test AUC = 0.836957513062
Test accuracy = 0.788012472126    
epoch 87: loss = 0.47712013577
Train AUC = 0.839958279695
Test AUC = 0.837148788666
Test accuracy = 0.788244152484    
epoch 88: loss = 0.477494554989
Train AUC = 0.840623544877
Test AUC = 0.835795499313
Test accuracy = 0.786525856493    
epoch 89: loss = 0.475754185156
Train AUC = 0.840596932381
Test AUC = 0.835039424644
Test accuracy = 0.786274869438    
epoch 90: loss = 0.476438145746
Train AUC = 0.840656867603
Test AUC = 0.837328204614
Test accuracy = 0.787481537971    
epoch 91: loss = 0.475515956228
Train AUC = 0.840037388623
Test AUC = 0.83634446248
Test accuracy = 0.787191937524    
epoch 92: loss = 0.473302981167
Train AUC = 0.841075017205
Test AUC = 0.83717760326
Test accuracy = 0.788398606057    
epoch 93: loss = 0.47263479865
Train AUC = 0.842028551779
Test AUC = 0.837295500216
Test accuracy = 0.787616684847    
epoch 94: loss = 0.472210624904
Train AUC = 0.841134904345
Test AUC = 0.837106780498
Test accuracy = 0.787520151365    
epoch 95: loss = 0.471078841975
Train AUC = 0.841882774923
Test AUC = 0.836676319554
Test accuracy = 0.787066443996    
epoch 96: loss = 0.471021121198
Train AUC = 0.841595380838
Test AUC = 0.835244732003
Test accuracy = 0.786236256045    
epoch 97: loss = 0.470454639558
Train AUC = 0.84066806758
Test AUC = 0.835746624055
Test accuracy = 0.786747883503    
epoch 98: loss = 0.470751455336
Train AUC = 0.840711739313
Test AUC = 0.835749052158
Test accuracy = 0.78670927011    
epoch 99: loss = 0.467939226916
Train AUC = 0.842460508814
Test AUC = 0.836335225131
Test accuracy = 0.787076097344    

# Synthetic
## No padding
epoch 0: loss = 0.696876761317
Train AUC = 0.524806690176
Test AUC = 0.62016711264
Test accuracy = 0.58769612711    
epoch 1: loss = 0.653329724073
Train AUC = 0.620341479704
Test AUC = 0.642495471272
Test accuracy = 0.627795431976    
epoch 2: loss = 0.637546107173
Train AUC = 0.653691561712
Test AUC = 0.661205015167
Test accuracy = 0.630297914598    
epoch 3: loss = 0.623001366854
Train AUC = 0.680261034038
Test AUC = 0.701584360563
Test accuracy = 0.654786494538    
epoch 4: loss = 0.60470367074
Train AUC = 0.711187095051
Test AUC = 0.726258400787
Test accuracy = 0.670158887786    
epoch 5: loss = 0.588938522339
Train AUC = 0.733415892172
Test AUC = 0.74134870886
Test accuracy = 0.681946375372    
epoch 6: loss = 0.571884593368
Train AUC = 0.757696155237
Test AUC = 0.76762625669
Test accuracy = 0.704597815293    
epoch 7: loss = 0.551027294993
Train AUC = 0.780747329675
Test AUC = 0.785378968105
Test accuracy = 0.717686196624    
epoch 8: loss = 0.539560818672
Train AUC = 0.792579529243
Test AUC = 0.794081262118
Test accuracy = 0.726812313803    
epoch 9: loss = 0.53293210566
Train AUC = 0.799477577572
Test AUC = 0.803603958849
Test accuracy = 0.735064548163    
epoch 10: loss = 0.524243408442
Train AUC = 0.807256010957
Test AUC = 0.810015735495
Test accuracy = 0.739384309831    
epoch 11: loss = 0.517234666646
Train AUC = 0.813468002103
Test AUC = 0.810274934583
Test accuracy = 0.738857994042    
epoch 12: loss = 0.514476633072
Train AUC = 0.815693275779
Test AUC = 0.813713164889
Test accuracy = 0.742423038729    
epoch 13: loss = 0.511241453886
Train AUC = 0.818626805642
Test AUC = 0.815289100034
Test accuracy = 0.746216484608    
epoch 14: loss = 0.506377921999
Train AUC = 0.822522680926
Test AUC = 0.818337884434
Test accuracy = 0.748480635551    
epoch 15: loss = 0.504567994177
Train AUC = 0.824036035578
Test AUC = 0.818130301249
Test accuracy = 0.748202581927    
epoch 16: loss = 0.502173940837
Train AUC = 0.825953830669
Test AUC = 0.819043577964
Test accuracy = 0.748728897716    
epoch 17: loss = 0.500695517659
Train AUC = 0.827035073296
Test AUC = 0.819618114428
Test accuracy = 0.749811320755    
epoch 18: loss = 0.498884834349
Train AUC = 0.828509883026
Test AUC = 0.818938638857
Test accuracy = 0.748897715988    
epoch 19: loss = 0.497651396692
Train AUC = 0.829490533421
Test AUC = 0.817424982061
Test accuracy = 0.747894736842    
epoch 20: loss = 0.497611442208
Train AUC = 0.829481328476
Test AUC = 0.817749568244
Test accuracy = 0.749175769613    
epoch 21: loss = 0.496288226545
Train AUC = 0.830561200288
Test AUC = 0.817269935799
Test accuracy = 0.748311817279    
epoch 22: loss = 0.4952380687
Train AUC = 0.83135808098
Test AUC = 0.817333427204
Test accuracy = 0.747567030785    
epoch 23: loss = 0.493331392109
Train AUC = 0.832797619327
Test AUC = 0.816102781262
Test accuracy = 0.746365441907    
epoch 24: loss = 0.491157665849
Train AUC = 0.83444433595
Test AUC = 0.816512339208
Test accuracy = 0.746434955313    
epoch 25: loss = 0.489196588099
Train AUC = 0.836007694796
Test AUC = 0.814933531466
Test accuracy = 0.745710029791    
epoch 26: loss = 0.487493427098
Train AUC = 0.83716773796
Test AUC = 0.813497451452
Test accuracy = 0.745968222443    
epoch 27: loss = 0.487096253037
Train AUC = 0.837511839687
Test AUC = 0.815329310701
Test accuracy = 0.74588877855    
epoch 28: loss = 0.48351457417
Train AUC = 0.840127539275
Test AUC = 0.812573711656
Test accuracy = 0.743247269116   

## Pad to 100
epoch 0: loss = 0.696876770258
Train AUC = 0.524806694574
Test AUC = 0.620167627208
Test accuracy = 0.58769612711    
epoch 1: loss = 0.650674960017
Train AUC = 0.625251240436
Test AUC = 0.647814807727
Test accuracy = 0.631578947368    
epoch 2: loss = 0.627244207263
Train AUC = 0.664773401413
Test AUC = 0.670623187574
Test accuracy = 0.637070506455    
epoch 3: loss = 0.609677547216
Train AUC = 0.702555855844
Test AUC = 0.713615041529
Test accuracy = 0.660099304866    
epoch 4: loss = 0.596700492501
Train AUC = 0.721681125425
Test AUC = 0.731279460268
Test accuracy = 0.673813306852    
epoch 5: loss = 0.581685858965
Train AUC = 0.745452109977
Test AUC = 0.753912779768
Test accuracy = 0.69816285998    
epoch 6: loss = 0.567390975356
Train AUC = 0.76351030391
Test AUC = 0.767837358323
Test accuracy = 0.707894736842    
epoch 7: loss = 0.554410547018
Train AUC = 0.778094326563
Test AUC = 0.784637765537
Test accuracy = 0.716742800397    
epoch 8: loss = 0.542615583539
Train AUC = 0.789506600251
Test AUC = 0.788883651222
Test accuracy = 0.720099304866    
epoch 9: loss = 0.535783621669
Train AUC = 0.796005204728
Test AUC = 0.799010338951
Test accuracy = 0.728411122145    
epoch 10: loss = 0.52872967124
Train AUC = 0.802924918687
Test AUC = 0.802025084651
Test accuracy = 0.731231380338    
epoch 11: loss = 0.522577613592
Train AUC = 0.808489684467
Test AUC = 0.808813474916
Test accuracy = 0.737904667329    
epoch 12: loss = 0.516133828461
Train AUC = 0.814315290319
Test AUC = 0.812704031432
Test accuracy = 0.743078450844    
epoch 13: loss = 0.513349305093
Train AUC = 0.816809080508
Test AUC = 0.813718778392
Test accuracy = 0.744697120159    
epoch 14: loss = 0.509841690958
Train AUC = 0.819695927671
Test AUC = 0.816216538337
Test accuracy = 0.747686196624    
epoch 15: loss = 0.50695373714
Train AUC = 0.822160769695
Test AUC = 0.814677327317
Test accuracy = 0.747567030785    
epoch 16: loss = 0.505538460612
Train AUC = 0.823403135926
Test AUC = 0.817884179743
Test accuracy = 0.748857994042    
epoch 17: loss = 0.503201811016
Train AUC = 0.825139013988
Test AUC = 0.815919311227
Test accuracy = 0.747159880834    
epoch 18: loss = 0.501104821265
Train AUC = 0.826749844183
Test AUC = 0.818011815313
Test accuracy = 0.749026812314    
epoch 19: loss = 0.499244329333
Train AUC = 0.828105858481
Test AUC = 0.816637149735
Test accuracy = 0.748043694141    
epoch 20: loss = 0.497219379246
Train AUC = 0.829737437287
Test AUC = 0.817353427876
Test accuracy = 0.747537239325    
epoch 21: loss = 0.495933508873
Train AUC = 0.830728392672
Test AUC = 0.816372914712
Test accuracy = 0.747596822244    
epoch 22: loss = 0.494788710773
Train AUC = 0.831544156317
Test AUC = 0.817153960108
Test accuracy = 0.747507447865    
epoch 23: loss = 0.492937120795
Train AUC = 0.83300943735
Test AUC = 0.816163778882
Test accuracy = 0.747398212512    
epoch 24: loss = 0.491531378031
Train AUC = 0.834011386065
Test AUC = 0.814350359937
Test accuracy = 0.746782522344    
epoch 25: loss = 0.489662836492
Train AUC = 0.835476976819
Test AUC = 0.814140595936
Test accuracy = 0.745143992056    
epoch 26: loss = 0.489619593322
Train AUC = 0.8354001456
Test AUC = 0.813094578619
Test accuracy = 0.744905660377    
epoch 27: loss = 0.488601993024
Train AUC = 0.836179907012
Test AUC = 0.813332835083
Test accuracy = 0.745471698113    
epoch 28: loss = 0.485555410385
Train AUC = 0.83849414151
Test AUC = 0.812701146642
Test accuracy = 0.744637537239    
epoch 29: loss = 0.483324725926
Train AUC = 0.84016716351
Test AUC = 0.811613233712
Test accuracy = 0.743952333664    
epoch 30: loss = 0.481054313481
Train AUC = 0.841828923504
Test AUC = 0.80982956845
Test accuracy = 0.743495531281    
epoch 31: loss = 0.479919619858
Train AUC = 0.842482611346
Test AUC = 0.809382288947
Test accuracy = 0.741300893744    
epoch 32: loss = 0.475410619378
Train AUC = 0.845764235808
Test AUC = 0.807478559903
Test accuracy = 0.740774577954    
epoch 33: loss = 0.475448657572
Train AUC = 0.84579242925
Test AUC = 0.807168355198
Test accuracy = 0.742512413108    
epoch 34: loss = 0.472338859737
Train AUC = 0.847913832757
Test AUC = 0.806464612839
Test accuracy = 0.739245283019    
epoch 35: loss = 0.468546713889
Train AUC = 0.850658858514
Test AUC = 0.805167403125
Test accuracy = 0.737994041708    
epoch 36: loss = 0.466397996247
Train AUC = 0.852152006096
Test AUC = 0.803832821199
Test accuracy = 0.737447864945    
epoch 37: loss = 0.461713384092
Train AUC = 0.855443685509
Test AUC = 0.801961198786
Test accuracy = 0.735898709037    
epoch 38: loss = 0.457487526536
Train AUC = 0.858326098777
Test AUC = 0.80131108955
Test accuracy = 0.735908639523    
epoch 39: loss = 0.455568157136
Train AUC = 0.859604380963
Test AUC = 0.799503721247
Test accuracy = 0.733843098312    
epoch 40: loss = 0.453459106386
Train AUC = 0.861037697659
Test AUC = 0.800363233352
Test accuracy = 0.733902681231    
epoch 41: loss = 0.446135167778
Train AUC = 0.865961623066
Test AUC = 0.798005648823
Test accuracy = 0.731857000993    
epoch 42: loss = 0.441914671659
Train AUC = 0.868622670349
Test AUC = 0.795810378318
Test accuracy = 0.730327706058    
epoch 43: loss = 0.440139184892
Train AUC = 0.869736383463
Test AUC = 0.79492647566
Test accuracy = 0.72912611718    
epoch 44: loss = 0.438669316471
Train AUC = 0.870485388012
Test AUC = 0.793494372573
Test accuracy = 0.729404170804    
epoch 45: loss = 0.433102045953
Train AUC = 0.874069603995
Test AUC = 0.793251777644
Test accuracy = 0.729682224429    
epoch 46: loss = 0.426854483783
Train AUC = 0.877905377832
Test AUC = 0.789942455711
Test accuracy = 0.725958291956    
epoch 47: loss = 0.424283537269
Train AUC = 0.879503717396
Test AUC = 0.78773137505
Test accuracy = 0.723048659384    
epoch 48: loss = 0.420851570368
Train AUC = 0.881453025749
Test AUC = 0.786335804649
Test accuracy = 0.722413108242    
epoch 49: loss = 0.423612777889
Train AUC = 0.879729095101
Test AUC = 0.786316761576
Test accuracy = 0.722581926514    
epoch 50: loss = 0.422539167106
Train AUC = 0.880467341491
Test AUC = 0.787847735448
Test accuracy = 0.724716981132    
epoch 51: loss = 0.412302298844
Train AUC = 0.886652856798
Test AUC = 0.783736857349
Test accuracy = 0.715710029791    
epoch 52: loss = 0.407349474728
Train AUC = 0.889489996085
Test AUC = 0.783695618969
Test accuracy = 0.717328699106    
epoch 53: loss = 0.402383060753
Train AUC = 0.892357981027
Test AUC = 0.783386745597
Test accuracy = 0.720943396226    
epoch 54: loss = 0.403081536293
Train AUC = 0.891740756335
Test AUC = 0.782751221878
Test accuracy = 0.719811320755    
epoch 55: loss = 0.402470844984
Train AUC = 0.89199298766
Test AUC = 0.782130989683
Test accuracy = 0.720625620655    
epoch 56: loss = 0.39227450639
Train AUC = 0.898149915409
Test AUC = 0.778887674886
Test accuracy = 0.716802383317    
epoch 57: loss = 0.391544473171
Train AUC = 0.898315982828
Test AUC = 0.776530059264
Test accuracy = 0.710923535253    
epoch 58: loss = 0.391136719286
Train AUC = 0.898437571197
Test AUC = 0.778912225175
Test accuracy = 0.717914597815    
epoch 59: loss = 0.394537183642
Train AUC = 0.896679704382
Test AUC = 0.775279540275
Test accuracy = 0.714428997021    
epoch 60: loss = 0.386353088915
Train AUC = 0.901061485208
Test AUC = 0.778115824239
Test accuracy = 0.717040714995    
epoch 61: loss = 0.376578217745
Train AUC = 0.906436186967
Test AUC = 0.775487830889
Test accuracy = 0.71551142006    
epoch 62: loss = 0.370892979205
Train AUC = 0.909388503663
Test AUC = 0.77685489135
Test accuracy = 0.71722939424    
epoch 63: loss = 0.370104502141
Train AUC = 0.909542075635
Test AUC = 0.775652695193
Test accuracy = 0.716703078451    
epoch 64: loss = 0.369716176391
Train AUC = 0.909965165951
Test AUC = 0.774637279214
Test accuracy = 0.714697120159    
epoch 65: loss = 0.366282129288
Train AUC = 0.911489774023
Test AUC = 0.7718184997
Test accuracy = 0.711737835154    
epoch 66: loss = 0.362041819096
Train AUC = 0.913679838722
Test AUC = 0.77168574792
Test accuracy = 0.713138033764    
epoch 67: loss = 0.360045430064
Train AUC = 0.914556358019
Test AUC = 0.769034863824
Test accuracy = 0.707000993049    
epoch 68: loss = 0.361317017674
Train AUC = 0.913933585566
Test AUC = 0.772269437471
Test accuracy = 0.712591857001    
epoch 69: loss = 0.367336970568
Train AUC = 0.910532952281
Test AUC = 0.770323904266
Test accuracy = 0.711727904667    
epoch 70: loss = 0.358976247907
Train AUC = 0.915079934395
Test AUC = 0.770160583665
Test accuracy = 0.709890764647    
epoch 71: loss = 0.37453148067
Train AUC = 0.906734916499
Test AUC = 0.768226158174
Test accuracy = 0.705799404171    
epoch 72: loss = 0.382613302767
Train AUC = 0.902517551035
Test AUC = 0.769335580524
Test accuracy = 0.710675273088    
epoch 73: loss = 0.362812243402
Train AUC = 0.913224770038
Test AUC = 0.76970160811
Test accuracy = 0.711966236346    
epoch 74: loss = 0.348231175542
Train AUC = 0.920436854712
Test AUC = 0.769967384398
Test accuracy = 0.71180734856    
epoch 75: loss = 0.340711933374
Train AUC = 0.923904292535
Test AUC = 0.768487197065
Test accuracy = 0.71140019861    
epoch 76: loss = 0.333513373137
Train AUC = 0.927579064647
Test AUC = 0.767576666338
Test accuracy = 0.709404170804    
epoch 77: loss = 0.329902178049
Train AUC = 0.928947956989
Test AUC = 0.766563455562
Test accuracy = 0.709116186693    
epoch 78: loss = 0.324799761176
Train AUC = 0.93134006265
Test AUC = 0.765138583046
Test accuracy = 0.709572989076    
epoch 79: loss = 0.321710561216
Train AUC = 0.932615041978
Test AUC = 0.764940072472
Test accuracy = 0.707715988083    
epoch 80: loss = 0.319352391362
Train AUC = 0.933544238624
Test AUC = 0.761490790136
Test accuracy = 0.706593843098    

## clean Assistments
epoch 0: loss = 0.650011429281
Train AUC = 0.647351183829
Test AUC = 0.676950107782
Test accuracy = 0.682796019308    
epoch 1: loss = 0.577679193381
Train AUC = 0.739154735792
Test AUC = 0.758721767332
Test accuracy = 0.732018354091    
epoch 2: loss = 0.539466670065
Train AUC = 0.773088880371
Test AUC = 0.77834994306
Test accuracy = 0.74629044753    
epoch 3: loss = 0.521693080664
Train AUC = 0.782247668944
Test AUC = 0.786922254024
Test accuracy = 0.751623860318    
epoch 4: loss = 0.510056951732
Train AUC = 0.794216574193
Test AUC = 0.788080167731
Test accuracy = 0.747705738633    
epoch 5: loss = 0.50583613551
Train AUC = 0.796200002163
Test AUC = 0.792134872607
Test accuracy = 0.754633216137    
epoch 6: loss = 0.497487285824
Train AUC = 0.803708960082
Test AUC = 0.797578163867
Test accuracy = 0.758759907038    
epoch 7: loss = 0.491444365545
Train AUC = 0.805246103422
Test AUC = 0.798813820789
Test accuracy = 0.75837256421    
epoch 8: loss = 0.484323868246
Train AUC = 0.808911624982
Test AUC = 0.799961717481
Test accuracy = 0.758178892795    
epoch 9: loss = 0.478937418172
Train AUC = 0.814238725252
Test AUC = 0.799327227823
Test accuracy = 0.758238484    
epoch 10: loss = 0.475033405152
Train AUC = 0.813650752094
Test AUC = 0.800486039672
Test accuracy = 0.759236636672    
epoch 11: loss = 0.470910218629
Train AUC = 0.815202441841
Test AUC = 0.799177561331
Test accuracy = 0.760547643168    
epoch 12: loss = 0.464220940164
Train AUC = 0.817062430712
Test AUC = 0.801430977988
Test accuracy = 0.759653775103    
epoch 13: loss = 0.459450728966
Train AUC = 0.819103802135
Test AUC = 0.801297142047
Test accuracy = 0.759996424528    
epoch 14: loss = 0.454177543069
Train AUC = 0.822267911889
Test AUC = 0.798095617712
Test accuracy = 0.758044812586    
epoch 15: loss = 0.451046900316
Train AUC = 0.821282837312
Test AUC = 0.798001170438
Test accuracy = 0.753769143674    
epoch 16: loss = 0.443816940893
Train AUC = 0.824709710148
Test AUC = 0.798414397954
Test accuracy = 0.758506644419    
epoch 17: loss = 0.440008179708
Train AUC = 0.825890783216
Test AUC = 0.799178272258
Test accuracy = 0.758625826828    
epoch 18: loss = 0.434786498547
Train AUC = 0.828843156746
Test AUC = 0.796767797545
Test accuracy = 0.756167689649    
epoch 19: loss = 0.429854905967
Train AUC = 0.832813650687
Test AUC = 0.796761346492
Test accuracy = 0.754782194148    
epoch 20: loss = 0.426911094875
Train AUC = 0.831894756106
Test AUC = 0.796999873845
Test accuracy = 0.754603420535    
epoch 21: loss = 0.422855591232
Train AUC = 0.834069879345
Test AUC = 0.797268940484
Test accuracy = 0.758506644419    
epoch 22: loss = 0.419082769842
Train AUC = 0.835894055363
Test AUC = 0.799744531711
Test accuracy = 0.757985221381    
epoch 23: loss = 0.414731755401
Train AUC = 0.837565988109
Test AUC = 0.795345099508
Test accuracy = 0.756361361063    
epoch 24: loss = 0.410961823933
Train AUC = 0.84064796968
Test AUC = 0.794229525576
Test accuracy = 0.755065252369    
epoch 25: loss = 0.408432859363
Train AUC = 0.839941968756
Test AUC = 0.796319655812
Test accuracy = 0.756376258864    
epoch 26: loss = 0.404511155504
Train AUC = 0.841404471841
Test AUC = 0.795369255729
Test accuracy = 0.75604850724    
epoch 27: loss = 0.403331611193
Train AUC = 0.843610952809
Test AUC = 0.790663758173
Test accuracy = 0.752264465765  

### pad in the start with 0,  pad to length of 50
epoch 0: loss = 0.676547610398
Train AUC = 0.591296362198
Test AUC = 0.615256836888
Test accuracy = 0.645342947381    
epoch 1: loss = 0.636413576025
Train AUC = 0.612674212394
Test AUC = 0.61898473441
Test accuracy = 0.646475180263    
epoch 2: loss = 0.627975624619
Train AUC = 0.616295300162
Test AUC = 0.634579985325
Test accuracy = 0.645253560574    
epoch 3: loss = 0.619352817535
Train AUC = 0.644243360006
Test AUC = 0.670009037179
Test accuracy = 0.659391573804    
epoch 4: loss = 0.606554123488
Train AUC = 0.672877344664
Test AUC = 0.669545180181
Test accuracy = 0.663950300936    
epoch 5: loss = 0.599973942294
Train AUC = 0.679943051599
Test AUC = 0.690003402524
Test accuracy = 0.674855491329    
epoch 6: loss = 0.591420999079
Train AUC = 0.699422341235
Test AUC = 0.692906147639
Test accuracy = 0.674870389131    
epoch 7: loss = 0.584437997052
Train AUC = 0.705209448986
Test AUC = 0.706730477185
Test accuracy = 0.685864966331    
epoch 8: loss = 0.57920409152
Train AUC = 0.713861387201
Test AUC = 0.714195561567
Test accuracy = 0.690423693463    
epoch 9: loss = 0.574340787801
Train AUC = 0.720412316479
Test AUC = 0.720124289369
Test accuracy = 0.698885644479    
epoch 10: loss = 0.568158724091
Train AUC = 0.728789354889
Test AUC = 0.72999012853
Test accuracy = 0.707466777904    
epoch 11: loss = 0.562560201594
Train AUC = 0.741465758534
Test AUC = 0.741387605453
Test accuracy = 0.718357070496    
epoch 12: loss = 0.556384982485
Train AUC = 0.750653576047
Test AUC = 0.747821243168
Test accuracy = 0.72129193731    
epoch 13: loss = 0.551075882984
Train AUC = 0.759515903171
Test AUC = 0.753615553183
Test accuracy = 0.724494964543    
epoch 14: loss = 0.547849215341
Train AUC = 0.760212117904
Test AUC = 0.757893281503
Test accuracy = 0.728994100471    
epoch 15: loss = 0.544333826412
Train AUC = 0.767502937829
Test AUC = 0.76218989445
Test accuracy = 0.731139383827    
epoch 16: loss = 0.54341306289
Train AUC = 0.771471166317
Test AUC = 0.762281193548
Test accuracy = 0.733746499017    
epoch 17: loss = 0.538296170307
Train AUC = 0.772683256292
Test AUC = 0.767234728513
Test accuracy = 0.737128299863    
epoch 18: loss = 0.535899128878
Train AUC = 0.776946371469
Test AUC = 0.769285444462
Test accuracy = 0.7390054228    
epoch 19: loss = 0.532053133755
Train AUC = 0.779350096445
Test AUC = 0.771158518502
Test accuracy = 0.739184196413    
epoch 20: loss = 0.530522759214
Train AUC = 0.781115490806
Test AUC = 0.774528643715
Test accuracy = 0.740539896311    
epoch 21: loss = 0.530535617561
Train AUC = 0.782325920073
Test AUC = 0.773076698672
Test accuracy = 0.74052499851    
epoch 22: loss = 0.526983867089
Train AUC = 0.783981537917
Test AUC = 0.774370022409
Test accuracy = 0.743966390561    
epoch 23: loss = 0.524398003564
Train AUC = 0.787074173867
Test AUC = 0.77849693237
Test accuracy = 0.744994338836    
epoch 24: loss = 0.521813328519
Train AUC = 0.788609976598
Test AUC = 0.778931440871
Test accuracy = 0.746171265121    
epoch 25: loss = 0.520655855085
Train AUC = 0.789717310769
Test AUC = 0.781574282562
Test accuracy = 0.748554913295    
epoch 26: loss = 0.517471237616
Train AUC = 0.791680202807
Test AUC = 0.78294802916
Test accuracy = 0.749344496752    
epoch 27: loss = 0.516126442136
Train AUC = 0.793007231566
Test AUC = 0.782215013393
Test accuracy = 0.746826768369    
epoch 28: loss = 0.516413295811
Train AUC = 0.792937108566
Test AUC = 0.783224250387
Test accuracy = 0.748480424289    
epoch 29: loss = 0.513334070191
Train AUC = 0.79525558868
Test AUC = 0.785313451443
Test accuracy = 0.750908765866    
epoch 30: loss = 0.510591638811
Train AUC = 0.797020233855
Test AUC = 0.787633468051
Test accuracy = 0.752115487754    
epoch 31: loss = 0.508337252971
Train AUC = 0.798751567226
Test AUC = 0.788072026812
Test accuracy = 0.752934866814    
epoch 32: loss = 0.506921568603
Train AUC = 0.799435319843
Test AUC = 0.788029115044
Test accuracy = 0.751534473512    
epoch 33: loss = 0.506540043788
Train AUC = 0.799572678773
Test AUC = 0.790089499197
Test accuracy = 0.752756093201    
epoch 34: loss = 0.504803994388
Train AUC = 0.801325352238
Test AUC = 0.790499520894
Test accuracy = 0.754111793099    
epoch 35: loss = 0.503747070378
Train AUC = 0.801619999904
Test AUC = 0.790695135987
Test accuracy = 0.753679756868    
epoch 36: loss = 0.503134256963
Train AUC = 0.800741654785
Test AUC = 0.791243737075
Test accuracy = 0.754230975508    
epoch 37: loss = 0.500725423748
Train AUC = 0.803430132881
Test AUC = 0.792381615273
Test accuracy = 0.755050354568    
epoch 38: loss = 0.498770762574
Train AUC = 0.804089681431
Test AUC = 0.792763152621
Test accuracy = 0.755244025982    
epoch 39: loss = 0.49816255136
Train AUC = 0.805655536227
Test AUC = 0.792102083287
Test accuracy = 0.755452595197    
epoch 40: loss = 0.496729507591
Train AUC = 0.805004250144
Test AUC = 0.793429927595
Test accuracy = 0.756078302843    
epoch 41: loss = 0.495895353231
Train AUC = 0.806697982172
Test AUC = 0.793542049492
Test accuracy = 0.755780346821    
epoch 42: loss = 0.495012210174
Train AUC = 0.80715534747
Test AUC = 0.794068758682
Test accuracy = 0.755959120434    
epoch 43: loss = 0.49379506346
Train AUC = 0.806528908812
Test AUC = 0.793495623492
Test accuracy = 0.754677909541    
epoch 44: loss = 0.49323955088
Train AUC = 0.808100962142
Test AUC = 0.79380016353
Test accuracy = 0.755810142423    
epoch 45: loss = 0.492035123435
Train AUC = 0.808430611884
Test AUC = 0.795479953803
Test accuracy = 0.756063405041    
epoch 46: loss = 0.490470832044
Train AUC = 0.80853836642
Test AUC = 0.794531627379
Test accuracy = 0.756644419284    
epoch 47: loss = 0.490922919729
Train AUC = 0.80893820427
Test AUC = 0.79527679491
Test accuracy = 0.757046659913    
epoch 48: loss = 0.488295282378
Train AUC = 0.810382080538
Test AUC = 0.795706464876
Test accuracy = 0.757597878553    
epoch 49: loss = 0.487139961936
Train AUC = 0.811338945491
Test AUC = 0.795066899215
Test accuracy = 0.756763601692    
epoch 50: loss = 0.486542218562
Train AUC = 0.810839839106
Test AUC = 0.795476538105
Test accuracy = 0.756227280853    
epoch 51: loss = 0.486422182936
Train AUC = 0.810187106555
Test AUC = 0.795002874463
Test accuracy = 0.756912579703    
epoch 52: loss = 0.485017738559
Train AUC = 0.812665220102
Test AUC = 0.796037361411
Test accuracy = 0.758000119182    
epoch 53: loss = 0.482742310473
Train AUC = 0.813998286864
Test AUC = 0.796450667755
Test accuracy = 0.757448900542    
epoch 54: loss = 0.481864620339
Train AUC = 0.814033326861
Test AUC = 0.796650048836
Test accuracy = 0.758491746618    
epoch 55: loss = 0.481298425884
Train AUC = 0.814297934281
Test AUC = 0.795866000089
Test accuracy = 0.757076455515    
epoch 56: loss = 0.481278913491
Train AUC = 0.813913914189
Test AUC = 0.797039534961
Test accuracy = 0.758268279602    
epoch 57: loss = 0.479320388852
Train AUC = 0.815911917197
Test AUC = 0.796809556952
Test accuracy = 0.758447053215    

# Assitments 124 skills
epoch 0: loss = 0.680330366799
Train AUC = 0.58598884455
Test AUC = 0.600679561668
Test accuracy = 0.632050165886    
epoch 1: loss = 0.645270085696
Train AUC = 0.6099977975
Test AUC = 0.614582329787
Test accuracy = 0.641297442293    
epoch 2: loss = 0.63453043049
Train AUC = 0.614028080019
Test AUC = 0.616316653988
Test accuracy = 0.639297395233    
epoch 3: loss = 0.627671248985
Train AUC = 0.615343956222
Test AUC = 0.624807802832
Test accuracy = 0.64259159039    
epoch 4: loss = 0.617752694722
Train AUC = 0.636694739508
Test AUC = 0.634704772926
Test accuracy = 0.633003129485    
epoch 5: loss = 0.607718167883
Train AUC = 0.667580508111
Test AUC = 0.677059676856
Test accuracy = 0.673615849785    
epoch 6: loss = 0.593942119317
Train AUC = 0.682009511008
Test AUC = 0.687879674568
Test accuracy = 0.676910044942    
epoch 7: loss = 0.584308002934
Train AUC = 0.702435313321
Test AUC = 0.703255224434
Test accuracy = 0.685616132144    
epoch 8: loss = 0.576963806694
Train AUC = 0.714962882419
Test AUC = 0.709740813455
Test accuracy = 0.693498670557    
epoch 9: loss = 0.572077885722
Train AUC = 0.723666520449
Test AUC = 0.721541697941
Test accuracy = 0.701592978658    
epoch 10: loss = 0.566663511775
Train AUC = 0.731992484241
Test AUC = 0.729218079106
Test accuracy = 0.706722511118    
epoch 11: loss = 0.560524029262
Train AUC = 0.742717164761
Test AUC = 0.737224254897
Test accuracy = 0.710934374926    
epoch 12: loss = 0.555958808372
Train AUC = 0.746518779311
Test AUC = 0.741081781285
Test accuracy = 0.714463869738    
epoch 13: loss = 0.551128903122
Train AUC = 0.755481472359
Test AUC = 0.748594061201
Test accuracy = 0.719840466835    
epoch 14: loss = 0.546371221542
Train AUC = 0.761470423045
Test AUC = 0.75274146159
Test accuracy = 0.723334666698    
epoch 15: loss = 0.542168936043
Train AUC = 0.765945765014
Test AUC = 0.757056204974
Test accuracy = 0.726758276665    
epoch 16: loss = 0.538969384901
Train AUC = 0.767642477969
Test AUC = 0.759336819471
Test accuracy = 0.727146521095    
epoch 17: loss = 0.535660443884
Train AUC = 0.772428001755
Test AUC = 0.763866660041
Test accuracy = 0.731746629332    
epoch 18: loss = 0.531618037007
Train AUC = 0.776026299908
Test AUC = 0.766590522129
Test accuracy = 0.733993741029    
epoch 19: loss = 0.529623745969
Train AUC = 0.776258207652
Test AUC = 0.76733373356
Test accuracy = 0.735217299231    
epoch 20: loss = 0.527214369088
Train AUC = 0.779648775929
Test AUC = 0.769909577264
Test accuracy = 0.736158497847    
epoch 21: loss = 0.524222578966
Train AUC = 0.779659584882
Test AUC = 0.771518755337
Test accuracy = 0.735052589473    
epoch 22: loss = 0.522344274954
Train AUC = 0.783176885579
Test AUC = 0.773015168794
Test accuracy = 0.738252664769    
epoch 23: loss = 0.519691920642
Train AUC = 0.78496506061
Test AUC = 0.774990178315
Test accuracy = 0.739993882209    
epoch 24: loss = 0.517525839986
Train AUC = 0.786207470099
Test AUC = 0.77587356095
Test accuracy = 0.739017388644    
epoch 25: loss = 0.515107692191
Train AUC = 0.787434437438
Test AUC = 0.776708669878
Test accuracy = 0.739676227676    
epoch 26: loss = 0.51294696602
Train AUC = 0.78906892378
Test AUC = 0.777444661661
Test accuracy = 0.740311536742    
epoch 27: loss = 0.51255388874
Train AUC = 0.787599388753
Test AUC = 0.778453696624
Test accuracy = 0.741617449822    
epoch 28: loss = 0.509414957328
Train AUC = 0.7912105967
Test AUC = 0.779570271406
Test accuracy = 0.741205675428    
epoch 29: loss = 0.509037019628
Train AUC = 0.790993696097
Test AUC = 0.77800805388
Test accuracy = 0.73978211252    
epoch 30: loss = 0.51037664576
Train AUC = 0.788399737084
Test AUC = 0.77913378295
Test accuracy = 0.740205651898    
epoch 31: loss = 0.507254879583
Train AUC = 0.792538246302
Test AUC = 0.780132343205
Test accuracy = 0.742193933975    
epoch 32: loss = 0.504677118677
Train AUC = 0.792926477201
Test AUC = 0.781269745166
Test accuracy = 0.742535118473    
epoch 33: loss = 0.503522396088
Train AUC = 0.794321271134
Test AUC = 0.78192126757
Test accuracy = 0.742229228923    
epoch 34: loss = 0.501661978888
Train AUC = 0.794708116534
Test AUC = 0.781804831556
Test accuracy = 0.7426527683    
epoch 35: loss = 0.501881346558
Train AUC = 0.793928490551
Test AUC = 0.783327522102
Test accuracy = 0.742817478058    
epoch 36: loss = 0.499477684498
Train AUC = 0.795922627578
Test AUC = 0.783136288699
Test accuracy = 0.744335160827    
epoch 37: loss = 0.498178711443
Train AUC = 0.79704050799
Test AUC = 0.78406459013
Test accuracy = 0.744711640274    
epoch 38: loss = 0.497669245258
Train AUC = 0.796418999798
Test AUC = 0.783492223652
Test accuracy = 0.74217040401    
epoch 39: loss = 0.495826074571
Train AUC = 0.7980716148
Test AUC = 0.784126495873
Test accuracy = 0.744593990447    
epoch 40: loss = 0.493808025664
Train AUC = 0.79894136606
Test AUC = 0.784910169834
Test accuracy = 0.745323419375    
epoch 41: loss = 0.492287464214
Train AUC = 0.799908975369
Test AUC = 0.785949299426
Test accuracy = 0.745511659098    
epoch 42: loss = 0.491757917585
Train AUC = 0.80011646491
Test AUC = 0.785266442759
Test accuracy = 0.745935198475    
epoch 43: loss = 0.491457974369
Train AUC = 0.799856820231
Test AUC = 0.785427654151
Test accuracy = 0.745994023389    
epoch 44: loss = 0.489717006683
Train AUC = 0.801107131144
Test AUC = 0.786500306806
Test accuracy = 0.746076378268    
epoch 45: loss = 0.488774076556
Train AUC = 0.801561952328
Test AUC = 0.786825585558
Test accuracy = 0.746664627403    
epoch 46: loss = 0.487253398606
Train AUC = 0.801054351077
Test AUC = 0.7856977043
Test accuracy = 0.746511682628    
epoch 47: loss = 0.486466314757
Train AUC = 0.802056257389
Test AUC = 0.786800542656
Test accuracy = 0.747841125674    
epoch 48: loss = 0.484713128119
Train AUC = 0.803036818054
Test AUC = 0.787091031785
Test accuracy = 0.747970540483    
epoch 49: loss = 0.483439644178
Train AUC = 0.803912215461
Test AUC = 0.78707412502
Test accuracy = 0.748252900068    
epoch 50: loss = 0.482611769077
Train AUC = 0.804163976968
Test AUC = 0.787520940227
Test accuracy = 0.74834701993    
epoch 51: loss = 0.481981032726
Train AUC = 0.803173197185
Test AUC = 0.787828435061
Test accuracy = 0.74808819031    
epoch 52: loss = 0.480106099085
Train AUC = 0.802846777597
Test AUC = 0.787837862571
Test accuracy = 0.747629355985    
epoch 53: loss = 0.478682620056
Train AUC = 0.805152795263
Test AUC = 0.787253705805
Test accuracy = 0.747629355985    
epoch 54: loss = 0.476968884468
Train AUC = 0.806552344297
Test AUC = 0.787781891457
Test accuracy = 0.747405821313    
epoch 55: loss = 0.476404353525
Train AUC = 0.806360660968
Test AUC = 0.787096750378
Test accuracy = 0.745546954046    
epoch 56: loss = 0.475482617364
Train AUC = 0.806673169487
Test AUC = 0.787906528358
Test accuracy = 0.746994046919    
epoch 57: loss = 0.474156050971
Train AUC = 0.807238646659
Test AUC = 0.788429209031
Test accuracy = 0.748076425328    
epoch 58: loss = 0.473062480038
Train AUC = 0.806908480385
Test AUC = 0.789083079085
Test accuracy = 0.748511729688    
epoch 59: loss = 0.469565828641
Train AUC = 0.809168889244
Test AUC = 0.787084654796
Test accuracy = 0.747358761383    
epoch 60: loss = 0.470481933969
Train AUC = 0.806417772174
Test AUC = 0.787473347609
Test accuracy = 0.746429327749    
epoch 61: loss = 0.467363907532
Train AUC = 0.809756326075
Test AUC = 0.788945259766
Test accuracy = 0.748123485258    
epoch 62: loss = 0.46700594371
Train AUC = 0.810337092003
Test AUC = 0.788239120254
Test accuracy = 0.747794065743    
epoch 63: loss = 0.464928649592
Train AUC = 0.810529338059
Test AUC = 0.789188000755
Test accuracy = 0.749170568719    
epoch 64: loss = 0.463495243679
Train AUC = 0.81124138021
Test AUC = 0.788033209429
Test accuracy = 0.746711687334    
epoch 65: loss = 0.463537920605
Train AUC = 0.811236036534
Test AUC = 0.789479084738
Test accuracy = 0.747923480552    
epoch 66: loss = 0.462210051941
Train AUC = 0.811430150236
Test AUC = 0.788121861549
Test accuracy = 0.745323419375    
epoch 67: loss = 0.461707016735
Train AUC = 0.810162475509
Test AUC = 0.788254060307
Test accuracy = 0.746158733147    
epoch 68: loss = 0.459145003196
Train AUC = 0.813387458296
Test AUC = 0.787644942333
Test accuracy = 0.747747005812    
epoch 69: loss = 0.458384979855
Train AUC = 0.811308763619
Test AUC = 0.786513171221
Test accuracy = 0.745899903527    
epoch 70: loss = 0.456440261819
Train AUC = 0.810153378158
Test AUC = 0.787314687497
Test accuracy = 0.744876350032    
epoch 71: loss = 0.455492246332
Train AUC = 0.81434861388
Test AUC = 0.78870283799
Test accuracy = 0.74721758159    
epoch 72: loss = 0.453732410164
Train AUC = 0.814707834506
Test AUC = 0.786535148472
Test accuracy = 0.746323442905    
epoch 73: loss = 0.452392905047
Train AUC = 0.814308367956
Test AUC = 0.787273011657
Test accuracy = 0.745829313631    
epoch 74: loss = 0.449206309788
Train AUC = 0.816179693586
Test AUC = 0.787873758093
Test accuracy = 0.747123461729    
epoch 75: loss = 0.452888779568
Train AUC = 0.810745002304
Test AUC = 0.785914015703
Test accuracy = 0.746146968164    
epoch 76: loss = 0.450561773596
Train AUC = 0.815241784934
Test AUC = 0.787496745129
Test accuracy = 0.746711687334    
epoch 77: loss = 0.448190538269
Train AUC = 0.816607500751
Test AUC = 0.787236556076
Test accuracy = 0.745546954046    
epoch 78: loss = 0.444724137133
Train AUC = 0.816674041988
Test AUC = 0.787190981305
Test accuracy = 0.746841102144    
epoch 79: loss = 0.444575596036
Train AUC = 0.8175364182
Test AUC = 0.786151929776
Test accuracy = 0.743970446363    
epoch 80: loss = 0.443356205117
Train AUC = 0.818397499458
Test AUC = 0.786910394421
Test accuracy = 0.74621755806    
epoch 81: loss = 0.442609824918
Train AUC = 0.819065371667
Test AUC = 0.783511991832
Test accuracy = 0.743111602626    
epoch 82: loss = 0.44115669619
Train AUC = 0.818224260042
Test AUC = 0.785319820234
Test accuracy = 0.744676345326    
epoch 83: loss = 0.439184499509
Train AUC = 0.818945919171
Test AUC = 0.785030187987
Test accuracy = 0.745288124426    
epoch 84: loss = 0.436072811033
Train AUC = 0.820621610435
Test AUC = 0.786955168891
Test accuracy = 0.746146968164    
epoch 85: loss = 0.434366461002
Train AUC = 0.821330978056
Test AUC = 0.785562651086
Test accuracy = 0.745405774254    
epoch 86: loss = 0.434443339254
Train AUC = 0.820318592193
Test AUC = 0.785045099901
Test accuracy = 0.743899856467    
epoch 87: loss = 0.432420207696
Train AUC = 0.821258161143
Test AUC = 0.785437812068
Test accuracy = 0.745794018683    
epoch 88: loss = 0.431072685755
Train AUC = 0.822546713543
Test AUC = 0.786749154889
Test accuracy = 0.746405797783    
epoch 89: loss = 0.428787631519
Train AUC = 0.822917252398
Test AUC = 0.784911527773
Test accuracy = 0.745535189063    
epoch 90: loss = 0.428769987641
Train AUC = 0.822285135524
Test AUC = 0.784431531712
Test accuracy = 0.743829266571    
epoch 91: loss = 0.428228254571
Train AUC = 0.822288595279
Test AUC = 0.785111110626
Test accuracy = 0.743899856467    
epoch 92: loss = 0.425700803598
Train AUC = 0.823594279925
Test AUC = 0.785257479998
Test accuracy = 0.744558695499    
epoch 93: loss = 0.425302867637
Train AUC = 0.824674390015
Test AUC = 0.784290172312
Test accuracy = 0.74434692581    
epoch 94: loss = 0.422514362769
Train AUC = 0.824968556233
Test AUC = 0.783766553063
Test accuracy = 0.74491164498    
epoch 95: loss = 0.42123366215
Train AUC = 0.82461267183
Test AUC = 0.785227992327
Test accuracy = 0.746017553354    
epoch 96: loss = 0.419960156535
Train AUC = 0.825933246749
Test AUC = 0.783445286403
Test accuracy = 0.743264547401    
epoch 97: loss = 0.419957580892
Train AUC = 0.825584777131
Test AUC = 0.782740831606
Test accuracy = 0.743676321796    
epoch 98: loss = 0.417801280816
Train AUC = 0.827777233385
Test AUC = 0.782468044703
Test accuracy = 0.743841031554    
epoch 99: loss = 0.414029469093
Train AUC = 0.82857114125
Test AUC = 0.782739271247
Test accuracy = 0.744158686087